website/docs/concepts/html/lists.md

```diff
@@ -70,9 +40,82 @@ html! {
     </ul>
 };
 ```
+
 <!--END_DOCUSAURUS_CODE_TABS-->
 
+## Keyed lists
+
+A keyed list is a very optimized list that has keys on all tags.
+A `key` gives an html element a unique identifier which Yew can use for optimization purposes.
+
+:::warn
+
+Key has to to be unique and has to not dependent on the order of the list.
+
+:::
+
+Generally it is recommended to always add keys to the lists.
+
+Most of the time, to add keys, you will need to write something like this:
+
+```rust , ignore
+use yew::{html};
+
+let names = vec!["Sam","Bob","Ray"]
+
+html! {
+    <div id="introductions">
+        {
+            names.into_iter().map(|name| {
+                html!{<div key={name}>{format!("Hello, I'am {}!",name)}</div>}
+            }).collect::<Html>()
+        }
+    </div>
+};
+
+```
+
+### Performance increases
+
+We have [Keyed list](https://github.com/yewstack/yew/tree/master/examples/keyed_list) example that lets you test the performance improvements, but here is rough example of testing:
+
+1. Go to [Keyed list](https://github.com/yewstack/yew/tree/master/examples/keyed_list) hosted demo
+2. Add 500 elements.
+3. Disable keys.
+4. Reverse the list.
+5. Look at "The last rendering took Xms" (At the time of writing this it was ~60ms)
+6. Enable keys.
+7. Reverse the list.
+8. Look at "The last rendering took Xms" (At the time of writing this it was ~30ms)
+
+So just at the time of writing this, for 500 components its a x2 increase of speed.
+
+### Detailed explanation
+
+Usually you just need a key on every list item when you iterate and the order of data can change.
+So lets say you iterate through `["bob","sam","rob"]` ended up with html:
+
+```html
+<div id="bob">My name is Bob</div>
+<div id="sam">My name is Sam</div>
+<div id="rob">My name is rob</div>
+```
+
+Then if your list changed to `["bob","rob"]`,
+yew would delete from previous html element with id="rob" and update id="sam" to be id="rob"
```

Comment:
Do you mean `key` here when you use `id`?

Reply:
Nope i meant html id